src/sentry/static/sentry/app/components/projects/bookmarkToggle.jsx

```diff
@@ -1,37 +1,61 @@
-import PropTypes from 'prop-types';
 import React from 'react';
+import Reflux from 'reflux';
+import classNames from 'classnames';
 
 import createReactClass from 'create-react-class';
 
 import ApiMixin from '../../mixins/apiMixin';
 
 import {update as projectUpdate} from '../../actionCreators/projects';
 
+import LatestContextStore from '../../stores/latestContextStore';
+
 const BookmarkToggle = createReactClass({
   displayName: 'BookmarkToggle',
 
-  propTypes: {
-    orgId: PropTypes.string.isRequired,
-    project: PropTypes.object.isRequired,
-  },
+  mixins: [ApiMixin, Reflux.listenTo(LatestContextStore, 'onLatestContextUpdate')],
 
-  mixins: [ApiMixin],
+  getInitialState() {
+    return {
+      orgId: null,
+      project: null,
```

Comment:
showing my ignorance of reflux, but do you need to pull the current state of the store out here in case it isn't updated after the component is mounted?

Reply:
It always "just works" since the component is mounted faster than the API calls return and trigger `onLatestContextUpdate`. I could add it to the initial state to be safe though?

Reply:
What about just using `connect`?

Reply:
That works. Updated for `Reflux.connect`